### ISSUE TITLE
OAS 2.0 recursive references aren't resolved

### ISSUE THREAD

**Issue by postatum** [aml-org/amf#441]
Using `amf-client-js@3.1.7`.

```js
const amf = require('amf-client-js')

const schema = `
{
  "swagger": "2.0",
  "definitions": {
    "Thing": {
      "type": "object",
      "additionalProperties": true,
      "required": [
        "a",
        "b"
      ],
      "properties": {
        "a": {
          "type": "string"
        },
        "b": {
          "type": "object",
          "additionalProperties": true,
          "properties": {
            "c": {
              "$ref": "#/definitions/Thing"
            }
          }
        }
      }
    }
  },
  "paths": {
    "/things/{thingId}": {
      "get": {
        "responses": {
          "200": {
            "schema": {"$ref": "#/definitions/Thing"}
          }
        }
      }
    }
  }
}
`

async function main () {
  await amf.AMF.init()
  const model = await amf.AMF.oas20Parser().parseStringAsync(schema)
  const resolved = await amf.AMF.resolveOas20(model)
  const generated = await amf.AMF.oas20Generator().generateString(resolved)
  console.log(generated)
}

main()
```

Produces

```json
{
  "swagger": "2.0",
  "paths": {
    "/things/{thingId}": {
      "get": {
        "responses": {
          "200": {
            "description": "",
            "schema": {
              "type": "object",
              "additionalProperties": true,
              "required": [
                "a",
                "b"
              ],
              "properties": {
                "a": {
                  "type": "string"
                },
                "b": {
                  "type": "object",
                  "additionalProperties": true,
                  "properties": {
                    "c": {
                      "$ref": "#/definitions/Thing"
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  },
  "info": {
    "title": "API",
    "version": "1.0"
  }
}
```

Note this part:
```json
                    "c": {
                      "$ref": "#/definitions/Thing"
                    }
```

**Comment by BlockLucas** [comment]
https://www.mulesoft.org/jira/browse/APIMF-1536

**Comment by BlockLucas** [comment]
Here the definition should be emitted as well, similar to the result of resolving JSON Schemas

**Comment by tomsfernandez** [comment]
Hi @postatum. I have tested this issue with amf version 4.1.2 JS distribution and it is working correctly. We have made many fixes since version 3.1.7.

The produced api using your code snippet is now:
```javascript
{
  "swagger": "2.0",
  "paths": {
    "/things/{thingId}": {
      "get": {
        "responses": {
          "200": {
            "schema": {
              "$ref": "#/definitions/Thing"
            },
            "description": ""
          }
        }
      }
    }
  },
  "info": {
    "title": "API",
    "version": "1.0"
  },
  "definitions": {
    "Thing": {
      "type": "object",
      "additionalProperties": true,
      "required": [
        "a",
        "b"
      ],
      "properties": {
        "a": {
          "type": "string"
        },
        "b": {
          "type": "object",
          "additionalProperties": true,
          "properties": {
            "c": {
              "$ref": "#/definitions/Thing"
            }
          }
        }
      }
    }
  }
}
```

**Comment by tomsfernandez** [comment]
I'm closing this issue as the problem has been resolved
